contracts/test/v0.8/foundry/vrf/VRFV2Wrapper_Migration.t.sol

```diff
@@ -0,0 +1,344 @@
+
+    // Deploy coordinator and consumer.
+    s_testCoordinator = new ExposedVRFCoordinatorV2_5(address(0));
+    s_wrapper = new VRFV2PlusWrapper(address(s_linkToken), address(s_linkNativeFeed), address(s_testCoordinator));
+    s_consumer = new VRFV2PlusWrapperConsumerExample(address(s_linkToken), address(s_wrapper));
+
+    // Configure the coordinator.
+    s_testCoordinator.setLINKAndLINKNativeFeed(address(s_linkToken), address(s_linkNativeFeed));
+    setConfigCoordinator(basicFeeConfig);
+    setConfigWrapper();
+
+    s_testCoordinator.s_config();
+
+    // Data structures for Migrateable Wrapper
+    s_newCoordinator = new VRFCoordinatorV2Plus_V2Example(address(0), address(s_testCoordinator));
+    vm.expectEmit(
+      false, // no first indexed topic
+      false, // no second indexed topic
+      false, // no third indexed topic
+      true // check data (target coordinator address)
+    );
+    address newCoordinatorAddr = address(s_newCoordinator);
+    emit CoordinatorRegistered(newCoordinatorAddr);
+    s_testCoordinator.registerMigratableCoordinator(newCoordinatorAddr);
+    assertTrue(s_testCoordinator.isTargetRegisteredExternal(newCoordinatorAddr));
+  }
+
+  function setConfigCoordinator(VRFCoordinatorV2_5.FeeConfig memory feeConfig) internal {
+    s_testCoordinator.setConfig(
+      0, // minRequestConfirmations
+      2_500_000, // maxGasLimit
+      1, // stalenessSeconds
+      50_000, // gasAfterPaymentCalculation
+      50000000000000000, // fallbackWeiPerUnitLink
+      feeConfig
+    );
+  }
+
+  function setConfigWrapper() internal {
+    s_wrapper.setConfig(
+      wrapperGasOverhead, // wrapper gas overhead
+      coordinatorGasOverhead, // coordinator gas overhead
+      0, // premium percentage
+      vrfKeyHash, // keyHash
+      10, // max number of words,
+      1, // stalenessSeconds
+      50000000000000000, // fallbackWeiPerUnitLink
+      0, // fulfillmentFlatFeeLinkPPM
+      0 // fulfillmentFlatFeeNativePPM
+    );
+    (
+      ,
+      ,
+      ,
+      uint32 _wrapperGasOverhead,
+      uint32 _coordinatorGasOverhead,
+      uint8 _wrapperPremiumPercentage,
+      bytes32 _keyHash,
+      uint8 _maxNumWords
+    ) = s_wrapper.getConfig();
+    assertEq(_wrapperGasOverhead, wrapperGasOverhead);
+    assertEq(_coordinatorGasOverhead, coordinatorGasOverhead);
+    assertEq(0, _wrapperPremiumPercentage);
+    assertEq(vrfKeyHash, _keyHash);
+    assertEq(10, _maxNumWords);
+  }
+
+  event RandomWordsRequested(
+    bytes32 indexed keyHash,
+    uint256 requestId,
+    uint256 preSeed,
+    uint256 indexed subId,
+    uint16 minimumRequestConfirmations,
+    uint32 callbackGasLimit,
+    uint32 numWords,
+    bytes extraArgs,
+    address indexed sender
+  );
+
+  function testMigrateWrapperLINKPayment() public {
+    s_linkToken.transfer(address(s_consumer), DEFAULT_LINK_FUNDING);
+
+    // Get type and version.
+    assertEq(s_wrapper.typeAndVersion(), "VRFV2Wrapper 1.0.0");
+
```

Comment:
can fund the subscription beforehand to make sure migration of funds work correctly and can assert funds no longer exist on v1 coordinator

Reply:
Done